larpandoracontent/LArTwoDReco/LArCosmicRay/TrackRefinementBaseAlgorithm.cc

```diff
@@ -1,215 +1,126 @@
-
-        if (!this->IsTrackContinuous(clusterAssociation, clusterToCaloHitListMap))
-            break;
-
-        this->CreateMainTrack(clusterAssociation, clusterToCaloHitListMap, pClusterList, microSlidingFitResultMap, macroSlidingFitResultMap, clusterVector);
-
-        mergeMade = true;
-    }
-    while(mergeMade && (loopIterations < m_maxMainLoopIterations));
-
-    return STATUS_CODE_SUCCESS;
+    m_maxHitSeparationForConnectedCluster(4.f),
+    m_maxTrackGaps(3),
+    m_lineSegmentLength(3.f),
+    m_hitWidthMode(true)
+{ 
 }
 
-//------------------------------------------------------------------------------------------------------------------------------------------
+//------------------------------------------------------------------------------------------------------------------------------------------    
 
-void TrackInEMShowerAlgorithm::SelectCleanClusters(const ClusterList *pClusterList, ClusterVector &clusterVector) const
+template<typename T>
+void TrackRefinementBaseAlgorithm::InitialiseContainers(const ClusterList *pClusterList, const T sortFunction, ClusterVector &clusterVector, SlidingFitResultMapPair &slidingFitResultMapPair) const
 {
+    const float slidingFitPitch(LArGeometryHelper::GetWireZPitch(this->GetPandora()));
+    
     for (const Cluster *const pCluster : *pClusterList)
     {
-        if (pCluster->GetNCaloHits() < m_minCaloHits)
+        if (LArClusterHelper::GetLengthSquared(pCluster) < m_minClusterLength * m_minClusterLength) 
             continue;
-
-        clusterVector.push_back(pCluster);
-    }
-
-    std::sort(clusterVector.begin(), clusterVector.end(), LArClusterHelper::SortByNHits);
-}
-
-//------------------------------------------------------------------------------------------------------------------------------------------
-
-void TrackInEMShowerAlgorithm::InitialiseSlidingFitResultMaps(const ClusterVector &clusterVector, TwoDSlidingFitResultMap &microSlidingFitResultMap,
-    TwoDSlidingFitResultMap &macroSlidingFitResultMap) const
-{
-    const float slidingFitPitch(LArGeometryHelper::GetWireZPitch(this->GetPandora()));
-
-    for (const Cluster *const pCluster : clusterVector)
-    {
+        
         try
         {
-            const TwoDSlidingFitResult microSlidingFitResult(pCluster, m_slidingFitWindow, slidingFitPitch);
+            const TwoDSlidingFitResult microSlidingFitResult(pCluster, m_microSlidingFitWindow, slidingFitPitch);
             const TwoDSlidingFitResult macroSlidingFitResult(pCluster, m_macroSlidingFitWindow, slidingFitPitch);
 
-            (void) microSlidingFitResultMap.insert(TwoDSlidingFitResultMap::value_type(pCluster, microSlidingFitResult));
-            (void) macroSlidingFitResultMap.insert(TwoDSlidingFitResultMap::value_type(pCluster, macroSlidingFitResult));
+            // ISOBEL: Need to check whether this is worth doing i.e. is it worth having to avoid shower clusters to try and recover small tracks?
```

Comment:
Presumably the conclusion is that this is worth retaining?

Reply:
actually no, it seems that its better to just reject all clusters below 50cm in order to avoid the shower contamination